test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -2087,6 +2087,39 @@ await pgMock
             Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
         }
 
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/3446")]
+        public async Task Bug3446()
```

Comment:
I'm so happy we have the mock for this kind of thing :)

Reply:
Yeah, it's an amazing thing.